pf/internal/check/checks.go

```diff
@@ -12,20 +12,44 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package tfgen
+package check
 
 import (
+	"errors"
 	"fmt"
 
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+
+	"github.com/pulumi/pulumi-terraform-bridge/pf/internal/muxer"
 	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
 	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
-	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
 )
 
-func checkIDProperties(sink diag.Sink, info tfbridge.ProviderInfo) error {
+// Validate that info is valid as either a PF provider or a PF & SDK based provider.
+//
+// This function should be called in the generate step, but before schema generation (so
+// as to error as soon as possible).
+func Provider(sink diag.Sink, info tfbridge.ProviderInfo) error {
+	isPFResource := func(string) bool { return true }
```

Comment:
Can we add a comment here that explains why we set the default to `true`? 
I presume "any provider that is _not_ muxed by default only contains PF resources so this always returns true for those".

Reply:
Added in 83fd9b4eb5b338c66ba9699b2cbdb12a31245315.